### DISCUSSION TITLE
[ 命名建議 ] LeetCode 876. Middle of the Linked List

### DISCUSSION THREAD

**Discussion by LPenny-github** [EngTW/English-for-Programmers#120] [Q&A] [answered]
LeetCode:
https://leetcode.com/problems/middle-of-the-linked-list/

線上教材:
https://blog.techbridge.cc/2019/11/22/leetcode-pattern-fast-and-slow-pointer/

---

這次一樣也改了方法的名字，再麻煩 @twy30 給予命名建議 orz 感激不盡

```csharp
public class Solution
{
    public ListNode GetMiddleNode(ListNode head)
    {
        ListNode heedNode = head;
        ListNode detector = heedNode, middleNote = heedNode;

        while (detector?.next != null)
        {
            detector = detector.next.next;
            middleNote = middleNote.next;
        }
        return middleNote;
    }
 }
```

**Comment by twy30** [comment, marked as answer]
@LPenny-github 

你好 😊

```C#
        ListNode detector = heedNode, middleNote = heedNode;
```

* `middleNote` 大概是 `middleNode` 的筆誤？
* `detector` 或許可以叫
  * `tailNode`
  * `tailNodeDetector`

其它的我覺得 OK 👌

**Reply by LPenny-github** [reply]
> `middleNote` 大概是 `middleNode` 的筆誤？

是的，我下次會再檢查一次 orz 

感謝 @twy30  大大喔

**Reply by LPenny-github** [reply]
> ListNode heedNode = head;

這邊也是筆誤，應該是 `headNode` orz
